File: dotnet/RailsLib.Net/Net/Common/ReportManager.cs
using GameLib.Net.Game;
using System;
using System.Collections.Generic;


/**
 * ReportManager controls the (non-action) communication with the client. 
 * 
 * Specific task:
 * Handling of the ReportBuffer and the DisplayBuffer
 */

namespace GameLib.Net.Common
{
    public class ReportManager : RailsManager
    {
        private readonly DisplayBuffer displayBuffer;
        private readonly ReportBuffer reportBuffer;


    private ReportManager(RailsRoot parent, string id) : base(parent, id)
        {
            displayBuffer = DisplayBuffer.Create(this, "displayBuffer");
            reportBuffer = ReportBuffer.Create(this, "reportBuffer");

            parent.StateManager.ChangeStack.AddChangeReporter(reportBuffer);
        }

        public static ReportManager Create(RailsRoot parent, string id)
        {
            return new ReportManager(parent, id);
        }

        public DisplayBuffer DisplayBuffer
        {
            get
            {
                return displayBuffer;
            }
        }

        public ReportBuffer ReportBuffer
        {
            get
            {
                return reportBuffer;
            }
        }
    }
}

File: src/main/java/net/sf/rails/common/ReportManager.java
package net.sf.rails.common;

import net.sf.rails.game.RailsManager;
import net.sf.rails.game.RailsRoot;

/**
 * ReportManager controls the (non-action) communication with the client.
 * <p>
 * Specific task:
 * Handling of the ReportBuffer and the DisplayBuffer
 */
public class ReportManager extends RailsManager {

    private final DisplayBuffer displayBuffer = new DisplayBuffer(this, "displayBuffer");

    private final ReportBuffer reportBuffer = new ReportBuffer(this, "reportBuffer");

    public ReportManager(RailsRoot parent, String id) {
        super(parent, id);

        parent.getStateManager().getChangeStack().addChangeReporter(reportBuffer);
    }

    public DisplayBuffer getDisplayBuffer() {
        return displayBuffer;
    }

    public ReportBuffer getReportBuffer() {
        return reportBuffer;
    }
}
